### PR TITLE
fix: issue with custom rule over inheritance

### PR DESCRIPTION
**Pull request by FlorianWoelki**
This is an improvement of #230 which fixes the #225 issue.

### CODE REVIEW

**Comment by Lisandra-dev** [comment]
Hello!
Seems to be a problem around useFilePath for rules.

Your implementation: 
image

Mine: 
image

In my implementation, I fixed this issue with 
```ts
  let hasIcon = plugin.getData()[file.path];
  if (typeof plugin.getData()[file.path] === 'object') {
    hasIcon = (plugin.getData()[file.path] as FolderIconObject).iconName;
  } else if (typeof plugin.getData()[file.path] === 'string') {
    hasIcon = plugin.getData()[file.path] as string;
  }
  ```

**Comment by FlorianWoelki** [comment]
> Hello! Seems to be a problem around useFilePath for rules.
> 
> Your implementation: image
> 
> Mine: image
> 
> In my implementation, I fixed this issue with
> 
> ```ts
>   let hasIcon = plugin.getData()[file.path];
>   if (typeof plugin.getData()[file.path] === 'object') {
>     hasIcon = (plugin.getData()[file.path] as FolderIconObject).iconName;
>   } else if (typeof plugin.getData()[file.path] === 'string') {
>     hasIcon = plugin.getData()[file.path] as string;
>   }
> ```
> 
> (here:
> 
> https://github.com/FlorianWoelki/obsidian-iconize/blob/c59765378bfcecd5f3df4974ca6b29ff544021a0/src/lib/customRule.ts#L131
> 
> )

How can I reproduce your icon structure? Is your custom rule applied to everything?

**Comment by Lisandra-dev** [comment]
The bug is caused because I have a subfolder with inheritance but without iconName, so in the settings:
```json
  "sandbox/subfolder/fix/eroerj er": {
    "iconName": null,
    "inheritanceIcon": "BoBxsLockOpenAlt"
  }
  ```
  
  The rules:
  ```json
        {
        "rule": "sandbox",
        "icon": "FarFolder",
        "for": "folders",
        "filepath": true,
        "useFilePath": true
      }
      ```

**Comment by FlorianWoelki** [comment]
> The bug is caused because I have a subfolder with inheritance but without iconName, so in the settings:
> 
> ```json
>   "sandbox/subfolder/fix/eroerj er": {
>     "iconName": null,
>     "inheritanceIcon": "BoBxsLockOpenAlt"
>   }
> ```
> 
> The rules:
> 
> ```json
>       {
>       "rule": "sandbox",
>       "icon": "FarFolder",
>       "for": "folders",
>       "filepath": true,
>       "useFilePath": true
>     }
> ```
> 
> TLDR, as folder can be considered to have an icon because it have the inheritance object, but … It doesn't have icon, as iconName is null!
> 
> Also, this one-liner can work too:
> 
> ```ts
> const hasIcon = typeof plugin.getData()[file.path] === 'object' ? 
>    (plugin.getData()[file.path] as FolderIconObject).iconName : 
>    plugin.getData()[file.path];
> ```

Feel free to check it again, should be fixed now.

**Comment by Lisandra-dev** [comment]
Work like a charm!

**Comment by FlorianWoelki** [comment]
> Work like a charm!

Awesome. I'll close the other PR for now, thanks for your work!

**Comment by Lisandra-dev** [comment]
Thank you too for the fix :)
